pkg/stash/with_azureblob.go

```diff
@@ -129,7 +129,7 @@ func (s *azblobLock) acquireLease(ctx context.Context, blobURL azblob.BlockBlobU
 	defer cancel()
 
 	// first we need to create a blob which can be then leased
-	_, err := blobURL.Upload(tctx, bytes.NewReader([]byte{1}), azblob.BlobHTTPHeaders{}, nil, azblob.BlobAccessConditions{})
+	_, err := blobURL.Upload(tctx, bytes.NewReader([]byte{1}), azblob.BlobHTTPHeaders{}, nil, azblob.BlobAccessConditions{}, azblob.DefaultAccessTier, nil, azblob.ClientProvidedKeyOptions{}, azblob.ImmutabilityPolicyOptions{})
```

Comment:
I'm somewhat concerned about `azblob.DefaultAccessTier` because I don't really trust that the people who built this SDK mean "default" as in "what it used to be before I implemented object level priority".

Reply:
In terms of the SDK and Azure it translates to this:
```
	// AccessTierNone represents an empty AccessTierType.
	AccessTierNone AccessTierType = ""
```

So, I think this is what users would expect who are existing users of Azure Blob Storage.

Reply:
It seems like the right choice: https://github.com/Azure/azure-storage-blob-go/commit/7b8190d57e7960b4f3fc6876bf3fc9bc614d26e2#diff-3316a810df43ef6f2e17497d349af1b78a9738ddd70bea1cfadc8f3e5b820f08L78